zerver/tests/test_bots.py

```diff
@@ -393,7 +395,11 @@ def test_bot_add_subscription(self) -> None:
             "principals": '["hambot-bot@zulip.testserver"]',
         }
         with self.capture_send_event_calls(expected_num_events=2) as events_bot:
-            result = self.common_subscribe_to_streams(hamlet, ["Rome"], bot_request_data)
+            result = self.common_subscribe_to_streams(
+                hamlet,
+                self.get_subscriptions_field_value(["Rome"]),
+                extra_post_data=bot_request_data,
+            )
```

Comment:
I don't love this interface. I think you could avoid almost all of the churn by having that `subscriptions` parameter be a list of `str | Dict[str, str]`, and if it's the former, `common_subscribe_to_streams` takes care of converting the format.

The benefit is that simple callers of `common_subscribe_to_streams` can remain very simple with just passing a list of strings if they don't have any other details to specify.

Reply:
Done